vql/darwin/xattr_test.go

```diff
@@ -0,0 +1,105 @@
+package darwin
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Velocidex/ordereddict"
+	"github.com/ivaxer/go-xattr"
+	"github.com/stretchr/testify/suite"
+	_ "www.velocidex.com/golang/velociraptor/accessors/file"
+	"www.velocidex.com/golang/velociraptor/file_store/test_utils"
+	"www.velocidex.com/golang/velociraptor/logging"
+	"www.velocidex.com/golang/velociraptor/services"
+	"www.velocidex.com/golang/velociraptor/vql/acl_managers"
+	"www.velocidex.com/golang/velociraptor/vtesting/assert"
+)
+
+type XattrTestSuite struct {
+	test_utils.TestSuite
+}
+
+func (self *XattrTestSuite) TestXAttr() {
+	filepath := "/tmp/xattr.test"
+	attr1 := "vr.test"
+	attr2 := "vr.test"
+
+	_, err := os.Create(filepath)
+	if err != nil {
+		self.T().Errorf("xattr: failed to create test file: %s", err)
+		return
+	}
+	defer os.Remove(filepath)
+
+	xattr.Set(filepath, attr1, []byte("test-value"))
+	xattr.Set(filepath, attr2, []byte("test-value"))
+
+	ctx := context.Background()
+	sub_ctx, cancel := context.WithTimeout(ctx, 100*time.Millisecond)
```

Comment:
100 ms seems very quick, let's give it a sec. Actually I don't think there are any cancellation points in this plugin at all so we don't really need a timeout